src/runtime/webgpu.cpp

```diff
@@ -0,0 +1,864 @@
+#include "HalideRuntimeWebGPU.h"
+#include "device_interface.h"
+#include "gpu_context_common.h"
+#include "printer.h"
+#include "scoped_spin_lock.h"
+
+// TODO: Discover this with CMake, and upstream toggle for stdint.h.
+#include "webgpu/webgpu.h"
+
+namespace Halide {
+namespace Runtime {
+namespace Internal {
+namespace WebGPU {
+
+extern WEAK halide_device_interface_t webgpu_device_interface;
+
+WEAK int create_webgpu_context(void *user_context);
+
+// A WebGPU instance/adapter/device defined in this module with weak linkage.
+WGPUInstance WEAK global_instance = nullptr;
```

Comment:
style nit: in Halide runtime, we generally prefer `WEAK type name;` to `name WEAK type;`

Reply:
Done. I had just copied that style from the equivalent variables in the CUDA and OpenCL runtimes which for some reason use `name WEAK type`, but I definitely prefer `WEAK type name` too.